rust_src/src/obarray.rs

```diff
@@ -60,9 +60,10 @@ impl LispObarrayRef {
     /// symbol with that name in this `LispObarrayRef`. If Emacs is loading Lisp
     /// code to dump to an executable (ie. `purify-flag` is `t`), the symbol
     /// name will be transferred to pure storage.
-    pub fn intern(&self, string: LispStringRef) -> LispObject {
+    pub fn intern(&self, string: impl Into<LispSymbolOrString>) -> LispObject {
         let string = string.into();
         let tem = self.lookup(string);
+        let string: LispObject = string.into();
```

Comment:
Instead of doing this here. What if the elseif/else below is re-written?

```
} else {
    let string_copy: LispObject = if unsafe { globals.Vpurify_flag }.is_not_nil() {
        // When Emacs is running lisp code to dump to an executable,
        // make use of pure storage.
        unsafe { Fpurecopy(string.into()) }
    } else {
        string.into()
    };
    intern_driver(string_copy, self.into(), tem)
}
```

Reply:
@agraven what are your thoughts on this refactor?

Reply:
After looking at it again it does seem nicer. I think I'll include it.